File: contest/goodbye2016/B.py
curr = 0  # North
flag = True 
for _ in range(input()):
    d, s = raw_input().split()
    if s == "North":
        if curr <= 0 :
            flag = False
        else:
            curr -= int(d)

    elif s == "South":
        if curr >= 20000:
            flag = False
        else:
            curr += int(d)
    elif s == "East" or s == "West":
        if curr >= 20000 or curr <=0 and int(d) > 0:
            flag = False

if flag :
    if curr == 0 :
        print "YES"
    else:
        print "NO"
else:
    print "NO"

File: contest/goodbye2016/B.cpp
#include<bits/stdc++.h>

using namespace std ;

int main(){

    int curr, t, d ; 
    bool flag = true;
    char s[10];
    curr = 0 ;
    flag = true ; 
    cin >> t;
    while(t--){
        cin >> d >> s ;
        if (s[0] == 'N'){
            if (curr <= 0){
                flag = false;
            }
            else{
                curr -= d ;
            }
        }

        else if(s[0] == 'S'){
            if (curr >= 20000){
                flag = false;
            }
            else{
                curr += d ;
            }
        }
        else if( s[0] == 'E' or s[0] == 'W'){
            if (curr >= 20000 or curr <=0 and d > 0){
                flag = false;
            }
        }

    }
    if(flag == true){
        if (curr == 0){
            cout  << "YES\n";
        }
        else{
            cout <<  "NO\n";
        }
    }
    else{
        cout << "NO\n" ;
    }
    

return 0 ;
}